astro/src/content/docs/extend/examples/modeling-organizations.mdx

```diff
@@ -0,0 +1,260 @@
+    }
+  }
+}
+```
+
+Extract the `id` from this JSON and store it in a variable.
+
+## Create Entity Grants
+
+The final step during the signup process will be to assign the correct permissions for the user to their Entity. To accomplish this, we will call the [Grant API](/docs/v1/tech/apis/entity-management/grants#grant-a-user-or-entity-permissions-to-an-entity). In order to call this API, we will need the Entity and User ids. Here's how the JSON will look for the grant request:
+
+```json
+{
+  "grant": {
+    "permissions": [
+      "Admin"
+    ],
+    "userId": "<user-id>"
+  }
+}
+```
+
+You'll notice that the JSON only includes the User id. The Entity id is added to the URL when calling this API like this:
+
+```http
+POST /api/entity/<user-id>/grant
+```
+
+It's important to note that the User id can be determine in a number of different ways, depending on how your registration process is set up. If you are using FusionAuth for registration, you can extract the User id from the JWT access token that FusionAuth provides at the end of the OAuth workflow. The user id is stored in the `sub` claim. Or you can use the [User Info](/docs/v1/tech/oauth/endpoints#userinfo) or [Introspect](/docs/v1/tech/oauth/endpoints#introspect) APIs with the access token to retrieve the user details. All of these places will contain the user id.
+
+Now we have constructed all the necessary pieces to model the customers of AwesomeCRM and the Users that have access to the customer accounts.
+
+## Login
+
+Now that our data model is prepared, we need to handle login events. When a user logs in, our application needs to know what organization they belong to. Generally, we store this information in a cookie or in a server-side session, such that it is available on each request to the application.
+
+To get this information from FusionAuth, we will call the [Search Grants](/docs/v1/tech/apis/entity-management/grants#search-for-grants) API. This API allows us to retrieve all of the grants that a specific User has to any Entities. This API doesn't take a JSON body, but we will supply the user id on the URL. The URL looks like this:
+
+```http
+GET /api/entity/grant/search?userId={userId}
+```
+
+We will replace the `userId` parameter with the id of the User that is currently logged in. In the same manner as above, this information can be retrieved from the access token JWT or the [User Info](/docs/v1/tech/oauth/endpoints#userinfo) or [Introspect](/docs/v1/tech/oauth/endpoints#introspect) APIs.
+
+The response from this API looks like this:
+
+```json
+{
+  "grants": [
+    {
+      "entity": {
+        "clientId": "092dbded-30af-4149-9c61-b578f2c72f59",
+        "clientSecret": "+fcXet9Iu2kQi61yWD9Tu4ReZ113P6yEAkr32v6WKOQ=",
+        "data": {
+          "companyType": "Legal"
+        },
+        "id": "8174f72f-5ecd-4eae-8de8-7fef597b3473",
+        "insertInstant": 1595361142909,
+        "lastUpdateInstant": 1595361143101,
+        "name": "Raviga",
```

Comment:
Not sure if this needs to be updated, but in the create example we used Pied Piper.

Reply:
I "think" this is the grant for a legal company called Raviga, that might belong to Pied?